physics/attach.go

```diff
@@ -15,39 +20,31 @@ func (v Vector) Vec() Vector {
 }
 
 // Attach takes in something for this vector to attach to and a set of
-// offsets. The resulting combined vector with the offsets is then returned,
-// and needs to be assigned to the calling vector.
-func (v Vector) Attach(a Attachable, offsets ...float64) Vector {
-	xOff := 0.0
-	yOff := 0.0
-	if len(offsets) > 0 {
-		xOff = offsets[0]
-	}
-	if len(offsets) > 1 {
-		yOff = offsets[1]
-	}
+// offsets.
+func (v *Vector) Attach(a Vecer, offX, offY float64) {
 	v2 := a.Vec()
-	v3 := Vector{v2.x, v2.y, xOff, yOff}
-	return v3
+	v.x = v2.x
+	v.y = v2.y
+	v.offX = offX
+	v.offY = offY
 }
 
 // AttachX performs an attachment that only attaches on the X axis.
-func (v Vector) AttachX(a Attachable, offX float64) Vector {
+func (v *Vector) AttachX(a Vecer, offX float64) {
 	v2 := a.Vec()
-	v3 := Vector{v2.x, v.y, offX, 0}
-	return v3
+	v.x = v2.x
+	v.offX = offX
 }
 
 // AttachY performs an attachment that only attaches on the Y axis.
-func (v Vector) AttachY(a Attachable, offY float64) Vector {
+func (v *Vector) AttachY(a Vecer, offY float64) {
 	v2 := a.Vec()
-	v3 := Vector{v.x, v2.y, 0, offY}
-	return v3
+	v.y = v2.y
+	v.offY = offY
 }
 
-// Detach returns a vector no longer attached to anything. The returned vector
-// needs to be assigned to the caller for the caller to be replaced (vectors
-// do not use pointer receivers)
-func (v Vector) Detach() Vector {
-	return NewVector(v.X(), v.Y())
+// Detach modified a vector to no longer be attached to anything.
+func (v *Vector) Detach() {
```

Comment:
Question: Since we are calling out the ability to Attach on a single axis should we also roll a DetachX DetachY?
It feels that we are trying to allow consumers of attach to be able to allow their attached vec to not be easily referenceable and the way to perform a DetachX in this set up would be to do something weird like make a newVec that relates to nothing and perform an AttachX on that new vec or perform a Detach followed by an AttachY on the original vec that you are attaching to.

Reply:
Created pull request to this branch to implement per our discussion.